README.md

```diff
@@ -134,10 +134,10 @@ By default, when building the project with no options, the following static libr
     -  iot_provisioning, iot_hub, etc.
   - az_storage_blobs
     -  Storage SDK blobs client.
-  - az_noplatform
+  - az_platform_stub
     - Library that provides a basic returning error for platform abstraction as AZ_NOT_IMPLEMENTED. This ensures the project can be compiled without the need to provide any specific platform implementation. This is useful if you want to use az_core without platform specific functions like `mutex` or `time`. 

```

Comment:
I think returning AZ_NOT_IMPLEMENTED at runtime is not a good idea, especially for boards where it may be hard to debug and find this. See above for one idea of breaking the build if the stub is included. 

I also think we should avoid PALs unless they are used (e.g. IoT should have no dependency on the PAL layer).

Reply:
Both of these points I think are part of a larger discussion which I am aggregating information for in parallel.

> (e.g. IoT should have no dependency on the PAL layer)

Definitely something we will be discussing. Not for this PR though.

Reply:
I prefer none.  What was Jeffrey's reason against none?

Reply:
paging @JeffreyRichter :)

Reply:
Could also be `az_platform_no_op`